gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/InstrumentatedGoogleHadoopFileSystem.java

```diff
@@ -327,4 +341,28 @@ public IOStatistics getIOStatistics() {
   public GhfsInstrumentation getInstrumentation() {
     return this.instrumentation;
   }
+
+  /** Set the Value for http get and head request related statistics keys */
+  private void setHttpStatistics() {
+    try {
+      GoogleCloudStorage gcs = getGcsFs().getGcs();
+      incrementStatistic(
+          GhfsStatistic.ACTION_HTTP_GET_REQUEST,
+          gcs.getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST).longValue());
+      incrementStatistic(
+          GhfsStatistic.ACTION_HTTP_HEAD_REQUEST,
+          gcs.getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_HEAD_REQUEST).longValue());
+      instrumentation.incrementFailureStatistics(
+          GhfsStatistic.ACTION_HTTP_GET_REQUEST.getSymbol(),
+          gcs.getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES)
+              .longValue());
+      instrumentation.incrementFailureStatistics(
+          GhfsStatistic.ACTION_HTTP_HEAD_REQUEST.getSymbol(),
+          gcs.getHttpStatistics(GoogleCloudStorageStatistics.ACTION_HTTP_HEAD_REQUEST_FAILURES)
+              .longValue());
```

Comment:
Do we need to store the statistics to gcs layer, instead can we fetch from gcsio when required ?

Reply:
Updated